discovery/targetgroup/targetgroup.go

```diff
@@ -91,3 +91,18 @@ func (tg *Group) UnmarshalJSON(b []byte) error {
 	tg.Labels = g.Labels
 	return nil
 }
+
+// MarshalYAML implements the json.Marshaler interface.
+func (tg Group) MarshalJSON() ([]byte, error) {
+	g := &struct {
+		Targets []string       `json:"targets"`
+		Labels  model.LabelSet `json:"labels,omitempty"`
+	}{
+		Targets: make([]string, 0, len(tg.Targets)),
+		Labels:  tg.Labels,
+	}
+	for _, t := range tg.Targets {
+		g.Targets = append(g.Targets, string(t[model.AddressLabel]))
```

Comment:
I copied this from MarshalYAML, but I wonder if this is even right. This would unexpectedly drop labels if your targetgroup has non-address labels within the target. Can that ever happen with a discoverer?

Reply:
Any non-address labels would be in the `labels` field of the target group, so this should be fine (it's basically just doing what the unmarshal methods do in reverse).